common_examples/iperf/main/iperf.c

```diff
@@ -0,0 +1,109 @@
+#include <string.h>
+#include "esp_netif.h"
+#include "esp_eth.h"
+#include "esp_event.h"
+#include "esp_console.h"
+#include "esp_log.h"
+#include "ethernet_init.h"
+#include "iperf_cmd.h"
+#include "sdkconfig.h"
+
+static const char *TAG = "iperf_example";
+
+static void start_dhcp_server_after_connection(void *arg, esp_event_base_t base, int32_t id, void *event_data)
+{
+    esp_netif_t *eth_netif = esp_netif_next_unsafe(NULL);
+    esp_eth_handle_t eth_handle = *(esp_eth_handle_t *)event_data;
+    while (eth_netif != NULL) {
+        esp_eth_handle_t eth_handle_for_current_netif = esp_netif_get_io_driver(eth_netif);
+        if (memcmp(&eth_handle, &eth_handle_for_current_netif, sizeof(esp_eth_handle_t)) == 0) {
+            esp_netif_dhcpc_stop(eth_netif);
+            esp_netif_dhcps_start(eth_netif);
+        }
+        eth_netif = esp_netif_next_unsafe(eth_netif);
+    }
+}
+
+void app_main(void)
+{
+    uint8_t eth_port_cnt = 0;
+    char if_key_str[10];
+    char if_desc_str[10];
+    esp_eth_handle_t *eth_handles;
+    esp_netif_config_t cfg;
+    esp_netif_inherent_config_t eth_netif_cfg;
+    esp_netif_init();
+    esp_event_loop_create_default();
+    ethernet_init_all(&eth_handles, &eth_port_cnt);
+
+#if CONFIG_EXAMPLE_ACT_AS_DHCP_SERVER
+    esp_netif_ip_info_t *ip_infos;
+
+    ip_infos = calloc(eth_port_cnt, sizeof(esp_netif_ip_info_t));
+
+    eth_netif_cfg = (esp_netif_inherent_config_t) {
+        .get_ip_event = IP_EVENT_ETH_GOT_IP,
+        .lost_ip_event = 0,
+        .flags = ESP_NETIF_DHCP_SERVER,
```

Comment:
```suggestion
        .flags = ESP_NETIF_DHCP_SERVER | ESP_NETIF_FLAG_AUTOUP,
```
If you use this `autoup` flag, you won't need to manually start the server in `start_dhcp_server_after_connection()`

Reply:
We've already discussed with Bogdan that usage of `ESP_NETIF_FLAG_AUTOUP` is not correct from system point of view since Ethernet interface is up only once the cable is connected. This is different to AP which can be considered up just from its start.

Reply:
should we fix this in `esp_netif`, then?

Reply:
Note that setting the Ethernet `netif` to up automatically at the start may have some unintended consequences when the `netif` considers interface up which is not up in reality. For example, `bridge` goes over all registered interfaces and forwards frames to that ones which are set up at `netif` layer. I don't remember the exact consequences but I faced this behavior when implementing the `bridge` and there was really some issue. If I remember correctly, I needed to fixed the Ethernet up/down handling in `netif` at that time to make it work...

Reply:
Do you mean systematic solution for DHCP server start? That's really kind of tricky right now. You have to start it exactly after connect event which may not be that obvious for users. It would be much more user friendly to start it at any time and it would wait for link up by itself. 
From just `netif` link up/down handling, I think it has been already address correctly at least for the Ethernet. Ethernet glue function `eth_action_connected` calls `esp_netif_action_connected` which sets the `netif` up and vice versa with disconnect event.